web/src/elements/types.ts

```diff
@@ -9,3 +10,66 @@ export type Constructor<T = object> = new (...args: any[]) => T;
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type AbstractConstructor<T = object> = abstract new (...args: any[]) => T;
+
```

Comment:
One thing that... irritates... me is that ak-dual-select does *not* adhere to this protocol.  It would be nice if all of our search-and-select tools did.  It's a YAGNI right now, but it's on my mind.